File: 1100/1327A/main.java
import java.util.*;
public class Main
{
    public static void main(String args[])
    {
        Scanner sc = new Scanner(System.in);
        int t = sc.nextInt();
        
        while(t-- > 0)
        {
            long n = sc.nextLong();
            long k = sc.nextLong();
            
            if((n&1) == (k&1) && n >= k*k)
                System.out.println("YES");
            else
                System.out.println("NO");
        }
    }
}

File: 1100/1327A/main.cpp
#include <iostream>
using namespace std;

int main() {
    int t;
    cin >> t;

    while (t--) {
        long long n, k;
        cin >> n >> k;

        if ((n % 2 == k % 2) && n >= k * k)
            cout << "YES" << endl;
        else
            cout << "NO" << endl;
    }

    return 0;
}
